WordPress/Classes/System/3DTouch/WP3DTouchShortcutCreator.swift

```diff
@@ -0,0 +1,114 @@
+    private let newPhotoPostShortcutIconImageName = "photos"
+    private let newPostShortcutIconImageName = "icon-new-post"
+    
+    override public init() {
+        application = UIApplication.sharedApplication()
+        mainContext = ContextManager.sharedInstance().mainContext
+        blogService = BlogService(managedObjectContext: mainContext)
+    }
+    
+    public func createShortcuts(loggedIn: Bool) {
+        if loggedIn {
+            createLoggedInShortcuts()
+        } else {
+            createLoggedOutShortcuts()
+        }
+    }
+    
+    public func loggedOutShortcutArray() -> [UIApplicationShortcutItem] {
+        let logInShortcut = UIMutableApplicationShortcutItem(type: WP3DTouchShortcutHandler.ShortcutIdentifier.LogIn.type,
+                                                   localizedTitle: NSLocalizedString("Log In", comment: "Log In 3D Touch Shortcut"),
+                                                localizedSubtitle: nil,
+                                                             icon: UIApplicationShortcutIcon(templateImageName: logInShortcutIconImageName),
+                                                                                                      userInfo: [WP3DTouchShortcutHandler.applicationShortcutUserInfoIconKey: WP3DTouchShortcutHandler.ShortcutIdentifier.LogIn.rawValue])
+        
+        return [logInShortcut]
+    }
+    
+    public func loggedInShortcutArray() -> [UIApplicationShortcutItem] {
+        var defaultBlogName: String?
+        if blogService.blogCountForAllAccounts() > 1 {
+            defaultBlogName = blogService.lastUsedOrFirstBlog().blogName
+        }
+        
+        let notificationsShortcut = UIMutableApplicationShortcutItem(type: WP3DTouchShortcutHandler.ShortcutIdentifier.Notifications.type,
+                                                           localizedTitle: NSLocalizedString("Notifications", comment: "Notifications 3D Touch Shortcut"),
+                                                        localizedSubtitle: nil,
+                                                                     icon: UIApplicationShortcutIcon(templateImageName: notificationsShortcutIconImageName),
+                                                                                                              userInfo: [WP3DTouchShortcutHandler.applicationShortcutUserInfoIconKey: WP3DTouchShortcutHandler.ShortcutIdentifier.Notifications.rawValue])
+        
+        let statsShortcut = UIMutableApplicationShortcutItem(type: WP3DTouchShortcutHandler.ShortcutIdentifier.Stats.type,
+                                                   localizedTitle: NSLocalizedString("Stats", comment: "Stats 3D Touch Shortcut"),
+                                                localizedSubtitle: defaultBlogName,
+                                                             icon: UIApplicationShortcutIcon(templateImageName: statsShortcutIconImageName),
```

Comment:
I actually messed up the formatting (userinfo parameter is not a part of the icon). I changed it so that it is correct and that helps a lot in readability. I will keep it this way as the parameters are all lined up now (and the lines are not as long)